tests/test_sagemaker/test_sagemaker_pipeline.py

```diff
@@ -0,0 +1,176 @@
+
+
+def test_list_pipelines_next_token(sagemaker_client):
+    fake_pipeline_names = ["APipelineName"]
+    _ = create_sagemaker_pipelines(sagemaker_client, fake_pipeline_names, 0.0)
+
+    response = sagemaker_client.list_pipelines(NextToken="0")
+    assert response["PipelineSummaries"].should.have.length_of(1)
+
+
+def test_list_pipelines_pipeline_name_prefix(sagemaker_client):
+    fake_pipeline_names = ["APipelineName", "BPipelineName", "CPipelineName"]
+    _ = create_sagemaker_pipelines(sagemaker_client, fake_pipeline_names, 0.0)
+    response = sagemaker_client.list_pipelines(PipelineNamePrefix="APipe")
+    assert response["PipelineSummaries"].should.have.length_of(1)
+    assert response["PipelineSummaries"][0]["PipelineName"].should.equal(
+        "APipelineName"
+    )
+
+    response = sagemaker_client.list_pipelines(PipelineNamePrefix="Pipeline")
+    assert response["PipelineSummaries"].should.have.length_of(3)
+
+
+def test_list_pipelines_created_after(sagemaker_client):
+    fake_pipeline_names = ["APipelineName", "BPipelineName", "CPipelineName"]
+    _ = create_sagemaker_pipelines(sagemaker_client, fake_pipeline_names, 0.0)
+
+    created_after_str = "2099-12-31 23:59:59"
+    response = sagemaker_client.list_pipelines(CreatedAfter=created_after_str)
+    assert response["PipelineSummaries"].should.be.empty
+
+    created_after_datetime = datetime.strptime(created_after_str, "%Y-%m-%d %H:%M:%S")
+    response = sagemaker_client.list_pipelines(CreatedAfter=created_after_datetime)
+    assert response["PipelineSummaries"].should.be.empty
+
+    created_after_timestamp = datetime.timestamp(created_after_datetime)
+    response = sagemaker_client.list_pipelines(CreatedAfter=created_after_timestamp)
+    assert response["PipelineSummaries"].should.be.empty
+
+
+def test_list_pipelines_created_before(sagemaker_client):
+    fake_pipeline_names = ["APipelineName", "BPipelineName", "CPipelineName"]
+    _ = create_sagemaker_pipelines(sagemaker_client, fake_pipeline_names, 0.0)
+
+    created_before_str = "2000-12-31 23:59:59"
+    response = sagemaker_client.list_pipelines(CreatedBefore=created_before_str)
+    assert response["PipelineSummaries"].should.be.empty
+
+    created_before_datetime = datetime.strptime(created_before_str, "%Y-%m-%d %H:%M:%S")
+    response = sagemaker_client.list_pipelines(CreatedBefore=created_before_datetime)
+    assert response["PipelineSummaries"].should.be.empty
+
+    created_before_timestamp = datetime.timestamp(created_before_datetime)
+    response = sagemaker_client.list_pipelines(CreatedBefore=created_before_timestamp)
+    assert response["PipelineSummaries"].should.be.empty
+
+
+def test_list_pipelines_invalid_values(sagemaker_client):
+    with pytest.raises(botocore.exceptions.ClientError):
+        _ = sagemaker_client.list_pipelines(MaxResults=200)  # Must be <= 100
+        _ = sagemaker_client.list_pipelines(NextToken="some-invalid-next-token")
```

Comment:
This doesn't work unfortunately. After the first command fails, we immediately jump out of the context, so the next three statements are ignored.

So this should either follow:
```
with pytest.raises():
    ....
with pytest.raises():
    ...
# etc
```
or we use `pytest.mark.parametrize` to execute the same test multiple times. I'll leave it up to you what works best!